client/material/button.tsx

```diff
@@ -171,47 +197,73 @@ export function useButtonState({
   )
 
   const keyDownActivatedRef = useRef(false)
-  const handleKeyDown = useCallback(
-    (event: React.KeyboardEvent) => {
-      const activationTarget = event.currentTarget
-      if (activationTarget.matches(':active')) {
-        rippleRef.current?.onActivate(event)
-        keyDownActivatedRef.current = true
-      } else {
-        keyDownActivatedRef.current = false
-
-        requestAnimationFrame(() => {
-          // Sometimes browsers don't active the element until after the event has been processed,
-          // so we check again on the next frame
-          if (activationTarget.matches(':active')) {
+  useKeyListener({
+    onKeyDown: useCallback(
+      (event: KeyboardEvent) => {
+        const activationTarget = event.currentTarget
+        if (activationTarget instanceof HTMLButtonElement) {
+          if (activationTarget?.matches(':active')) {
             rippleRef.current?.onActivate(event)
             keyDownActivatedRef.current = true
+          } else {
+            keyDownActivatedRef.current = false
+
+            requestAnimationFrame(() => {
+              // Sometimes browsers don't active the element until after the event has been
+              // processed, so we check again on the next frame
+              if (activationTarget?.matches(':active')) {
+                rippleRef.current?.onActivate(event)
+                keyDownActivatedRef.current = true
+              }
+            })
           }
-        })
-      }
+        }
+
+        if (
```

Comment:
As mentioned above, split the hotkey handling back out of this, into a handler that it *only* passed to the keylistener, and leave the previous keyboard handling intact as a React event handler attached to the button. No real reason to do both in the same handler.